packages/components/src/components/VirtualizedList/VirtualizedList.tsx

```diff
@@ -48,133 +48,179 @@ const Item = styled.div`
 
-                setStartIndex(newStartIndex);
-                setEndIndex(newEndIndex);
+            setStartIndex(newStartIndex);
+            setEndIndex(newEndIndex);
 
-                if (newEndIndex >= items.length - LOAD_MORE_BUFFER_COUNT) {
-                    debouncedOnScrollEnd();
-                }
-                onScroll?.(e);
-            },
-            [containerRef, debouncedOnScrollEnd, itemHeights, items.length, onScroll],
-        );
+            if (newEndIndex >= items.length - LOAD_MORE_BUFFER_COUNT) {
+                debouncedOnScrollEnd();
+            }
+            onScroll?.(e);
+        },
+        [containerRef, debouncedOnScrollEnd, itemHeights, items.length, onScroll],
+    );
 
-        useEffect(() => {
-            const container = containerRef.current;
-            if (container) {
-                container.addEventListener('scroll', handleScroll);
+    useEffect(() => {
+        const container = containerRef.current;
+        if (container) {
+            container.addEventListener('scroll', handleScroll);
 
-                return () => container.removeEventListener('scroll', handleScroll);
-            }
-        }, [containerRef, handleScroll]);
-
-        return (
-            <Container ref={containerRef} $height={listHeight} $minHeight={listMinHeight}>
-                <Content style={{ height: `${totalHeight}px` }}>
-                    {itemHeights.slice(startIndex, endIndex).map((height, index) => {
-                        const itemIndex = startIndex + index;
-                        const itemTop = itemHeights
-                            .slice(0, itemIndex)
-                            .reduce((acc, h) => acc + h, 0);
-
-                        if (!items[itemIndex]) return null;
-
-                        return (
-                            <Item
-                                key={itemIndex}
-                                style={{
-                                    top: `${itemTop}px`,
-                                    height,
-                                }}
-                            >
-                                {renderItem(items[itemIndex], itemIndex)}
-                            </Item>
-                        );
-                    })}
-                </Content>
-            </Container>
-        );
-    },
-);
+            return () => container.removeEventListener('scroll', handleScroll);
+        }
+    }, [containerRef, handleScroll]);
+
+    return (
+        <ListContainer<T>
+            ref={containerRef}
+            listHeight={listHeight}
+            listMinHeight={listMinHeight}
+            totalHeight={totalHeight}
+            items={items}
+            itemHeights={itemHeights}
+            startIndex={startIndex}
+            endIndex={endIndex}
+            renderItem={renderItem}
+        />
+    );
+}
+
+// NOTE: typecast here + memo() and forwardRef() because of passing the ref for useShadow()
+export const VirtualizedList = memo(
```

Comment:
Unfortunatelly, `memo()` and `forwardRef()` eat the generic. This can be bypassed with typecast, but that kills passing of the ref. I think it should be possible to create some type helper for it, altough I am not sure if it makes sense here to dive that deep.

Reply:
Pls advice, if you have any idea why direct passing of the `ref` prop doesn't work and I am forced to use the `forwardRef` again, thanks

Reply:
Sorry, no idea 🤷‍♂️

Reply:
This could have been, it, but it doesnt matter anymore.